test/token/ERC721/extensions/ERC721Consecutive.test.js

```diff
@@ -1,222 +1,244 @@
-  ];
-  const delegates = [user1, user3];
-
-  for (const offset of [0, 1, 42]) {
+
+const name = 'Non Fungible Token';
+const symbol = 'NFT';
+
+describe('ERC721Consecutive', function () {
+  for (const offset of [0n, 1n, 42n]) {
     describe(`with offset ${offset}`, function () {
-      beforeEach(async function () {
-        this.token = await ERC721ConsecutiveMock.new(
+      async function fixture() {
+        const accounts = await ethers.getSigners();
+        const [alice, bruce, chris, receiver] = accounts;
+
+        const batches = [
+          { receiver: alice, amount: 0n },
+          { receiver: alice, amount: 1n },
+          { receiver: alice, amount: 2n },
+          { receiver: bruce, amount: 5n },
+          { receiver: chris, amount: 0n },
+          { receiver: alice, amount: 7n },
+        ];
+        const delegates = [alice, chris];
+
+        const token = await ethers.deployContract('$ERC721ConsecutiveMock', [
           name,
           symbol,
           offset,
           delegates,
           batches.map(({ receiver }) => receiver),
           batches.map(({ amount }) => amount),
-        );
+        ]);
+
+        return { accounts, alice, bruce, chris, receiver, batches, delegates, token };
+      }
+
+      beforeEach(async function () {
+        Object.assign(this, await loadFixture(fixture));
       });
 
       describe('minting during construction', function () {
         it('events are emitted at construction', async function () {
           let first = offset;
-
-          for (const batch of batches) {
+          for (const batch of this.batches) {
             if (batch.amount > 0) {
-              await expectEvent.inConstruction(this.token, 'ConsecutiveTransfer', {
-                fromTokenId: web3.utils.toBN(first),
-                toTokenId: web3.utils.toBN(first + batch.amount - 1),
-                fromAddress: constants.ZERO_ADDRESS,
-                toAddress: batch.receiver,
-              });
+              await expect(this.token.deploymentTransaction())
+                .to.emit(this.token, 'ConsecutiveTransfer')
+                .withArgs(
+                  first /* fromTokenId */,
+                  first + batch.amount - 1n /* toTokenId */,
+                  ethers.ZeroAddress /* fromAddress */,
+                  batch.receiver.address /* toAddress */,
+                );
             } else {
-              // expectEvent.notEmitted.inConstruction only looks at event name, and doesn't check the parameters
+              // ".to.not.emit" doesn't check the parameters
+              // await expect(this.token.deploymentTransaction())
+              //   .to.not.emit(this.token, 'ConsecutiveTransfer')
+              //   .withArgs(
+              //     first, /* fromTokenId */
+              //     first + batch.amount - 1n, /* toTokenId */
+              //     ethers.ZeroAddress, /* fromAddress */
+              //     batch.receiver, /* toAddress */
+              //   );
```

Comment:
Is there any reason for keeping this? I guess we previously kept it hoping there would be a way to check for it

Reply:
Yes, that is what I was thinking

Reply:
Ok, let's drop it